### PR TITLE
Fixed critical issues

### PR DESCRIPTION
**Pull request by NullSoldier**
I wanted to use your library but when I pulled your code down it didn't even compile. On top of that, reporting page views didn't work at all. I also noticed you removed events from your example so I re-added it back in.

Another major change is that I've removed the dependency of the app.config file. If you use TrackMe, it will still look in your app.config file for the account code; but otherwise if you invoke the library directly (which is what I wanted) then you can now specify the Google account code.

Critical fixes:
- Page view requests were not sent correctly causing them to never be reported
- The project did not compile because of the sample

Improvements:
- Removed dependency on app.config file
- Added derived files to .gitignore so they won't be added to the repository
- Removed lots of unused code, there's no reason for it. You can get it from your GIT history if you need it again
- Since the data objects (GooglePageView and GoogleEvent) are immutable (and should be), there's no need for validation functions. I moved validation to the constructor.
- Lots of XML comments referenced non-existent parameters. I removed them

In short, I fixed lots of things. The example now works, the code now works. I've tested it.

### CODE REVIEW

**Comment by NullSoldier** [comment]
I also found two more bugs in my refactoring. As soon as you review/approve this pull request I would like to submit another one with fixes in TrackRequest for events, and page titles with spaces in the name.

**Comment by dougrathbone** [comment]
hi there,

appreciate your input. I'm a bit snowed under, but will be able to review this this weekend. If nothing was working, was it related to the version of the analytic api in use (the last pull request changed it)...?

**Comment by NullSoldier** [comment]
Unfortunately, I did not git blame to find out when it was broken. I'm
guessing it was written for an older version of the API and the last person
updated it. Either way, mine fixes it.

I did notice that there were a few more bugs that I fixed as well. (I broke
events in my pull request). I also changed the API so it can easily work in
a non ASP.net project without relying on an app config file. It shouldn't
break the API for anyone using the TrackMe. (It'll still grab it from the
app config in that case).

I'll submit another pull request to fix breaking events and another bug
after you review this one.
